src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -836,6 +841,52 @@ def perform_checks(self, analyze_ctx: AnalyzeContext) -> dict[str, CheckResult]:
 
         return results
 
+    def _resolve_more_dependencies(
+        self, dependencies: dict[str, DependencyInfo], context: AnalyzeContext, description: str
+    ) -> None:
+        """Utilise the Repo Finder to resolve more dependencies to repositories."""
+        purl = PackageURL.from_string(context.component.purl)
+        build_tools = context.dynamic_data["build_spec"]["tools"]
+        if not purl.type or not build_tools:
+            logger.debug("No PURL type found for: %s", purl)
+        else:
+            # Get actual PURL type based on current build tool
+            build_tool = build_tools[0]
+            if len(build_tools) > 1:
+                logger.debug(
+                    "More than one build tool found for %s, assuming type based on class: %s",
+                    description,
+                    build_tool.__class__,
+                )
+            updated_type = ""
+
+            if isinstance(build_tool, Gradle):
```

Comment:
I can't seem to figure out the reason why we need to "reconstruct" the purl of the software component here instead of using it directly?

Reply:
The purl here is of the main target, meaning its type is github or similar. A new purl has to be created for the dependency using its package type.

Reply:
Right. So we use the main's package type (e.g `Maven`) to resolve all of its dependencies by "casting" the dependencies PURL into that particular package type.